corehq/toggles.py

```diff
@@ -265,6 +265,43 @@ def enabled(self, item, namespace=Ellipsis):
             or super(PredictablyRandomToggle, self).enabled(item, namespace)
         )
 
+
+class DynamicallyPredictablyRandomToggle(PredictablyRandomToggle):
+    """
+    A PredictablyRandomToggle whose randomness can be configured via the database/UI.
+    """
+    RANDOMNESS_KEY = 'hq_dynamic_randomness'
+
+    def __init__(
+        self,
+        slug,
+        label,
+        tag,
+        namespaces,
+        default_randomness=0,
+        help_link=None,
+        description=None,
+        always_disabled=None
+    ):
+        super(PredictablyRandomToggle, self).__init__(slug, label, tag, list(namespaces),
+                                                      help_link=help_link, description=description,
+                                                      always_disabled=always_disabled)
+        assert namespaces, 'namespaces must be defined!'
```

Comment:
not blocking. I think you sent us a talk about instagram scaling up and they started using python -O which gets rid of asserts. Should we stop using these and start throwing exceptions?

Reply:
+1

Reply:
@snopoke didn't realize that. Should this be communicated to the team to no longer use `assert` ? I don't see a way to enforce this using stickler (flake8/pep8)

Reply:
I think it's bad practice to use `assert` for anything that get's relied on in production. In this case though I actually think it's fine since these classes are statically defined in code so the `assert` would fail locally or during build. If this `assert` got removed in production by using `-O` that won't affect anything.

Reply:
from a performance perspective, the reason you compile asserts out is just to avoid the branching logic right? so if we had `if not namespaces: raise Exception()` would that not be the same? 

I'm happy to swap it (and the code it was copied from) to use `Exception`, but agree with @emord that we should communicate out to the team if that's the suggested standard moving forwards.

Reply:
@czue correct re branching logic but what I was saying is that you shouldn't really use asserts for runtime validation since they can get compiled out. In this case it's fine since they aren't really runtime validations.